app/src/main/java/net/gsantner/opoc/frontend/base/GsFragmentBase.java

```diff
@@ -149,17 +128,33 @@ public String getAppLanguage() {
     public void onFragmentFirstTimeVisible() {
     }
 
-    @Override
-    public void setUserVisibleHint(boolean isVisibleToUser) {
-        super.setUserVisibleHint(isVisibleToUser);
-        synchronized (_fragmentFirstTimeVisibleSync) {
-            if (isVisibleToUser && _fragmentFirstTimeVisible) {
-                _fragmentFirstTimeVisible = false;
-                onFragmentFirstTimeVisible();
-            }
+    private synchronized void checkRunFirstTimeVisible() {
+        if (_fragmentFirstTimeVisible && isVisible() && isResumed()) {
+            _fragmentFirstTimeVisible = false;
+            onFragmentFirstTimeVisible();
+            attachToolbarClickListenersToFragment();
         }
+    }
 
-        attachToolbarClickListenersToFragment();
+    protected void attachToolbarClickListenersToFragment() {
+        final Toolbar toolbar = getToolbar();
+        if (toolbar != null) {
+            toolbar.setOnLongClickListener(clickView -> isVisible() && isResumed() && onToolbarLongClicked(clickView));
```

Comment:
suggestion: Callback.b2<bool,view> (islong, view) {

if ( isvisible && isRes.....) {


  if islong { return ontoolbarlongclick() }

  else { ontoolbarclick }
}
return false

---


setOnClivkListener(v -> callback(false, v));

setOnLonglivkListener(v -> callback(true, v));

what do you think?

Reply:
IMO it's extra indirection for no real benefit.

Reply:
its easier to debug as you can put a break point on i.e. isVisible line , to add i.e. another condition to both clicks